content/guides/dapps/journal.md

```diff
@@ -0,0 +1,564 @@
+}
+```
+
+In the above code, we used the following macros:
+
+- `#[derive(Accounts)]` macro is used to deserialize and validate the list of
+  accounts specified within the struct.
+- `#[instruction(...)]` attribute macro is used to access the instruction data
+  passed into the instruction.
+- `#[account(...)]` attribute macro then specifies additional constraints on the
+  accounts.
+
+A journal entry is a
+[PDA](https://solanacookbook.com/core-concepts/pdas.html#facts). Since we are
+creating a new journal entry, it needs to be initialized.
+
+A PDA is initalized with `seeds`, `bumps`, and `init`. The `init` constraint
+also requires a `payer` and `space` to define who is paying the rent to hold
+this account on chain and how much space needs to be allocated.
+
+Note: Calculating space is defined
+[here](https://book.anchor-lang.com/anchor_references/space.html).
+
+### Updating a journal entry
+
+Next, lets add an `update_journal_entry` instruction with a context that has an
+`UpdateEntry` type.
+
+To do this, the instruction will need rewrite the data for a specific PDA that
+was saved to the `JournalEntryState` of the account when the owner of
+the journal entry calls the `update_journal_entry` instruction.
+
+```rust
+#[program]
+mod journal {
+    use super::*;
+
+    ...
+
+    pub fn update_journal_entry(
+        ctx: Context<UpdateEntry>,
+        title: String,
+        message: String,
+    ) -> Result<()> {
+        msg!("Journal Entry Updated");
+        msg!("Title: {}", title);
+        msg!("Message: {}", message);
+
+        let journal_entry = &mut ctx.accounts.journal_entry;
+        journal_entry.message = message;
+
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+#[instruction(title: String, message: String)]
+pub struct UpdateEntry<'info> {
+    #[account(
+        mut,
+        seeds = [title.as_bytes(), owner.key().as_ref()],
+        bump,
+        realloc = 8 + 32 + 1 + 4 + title.len() + 4 + message.len(),
+        realloc::payer = owner,
+        realloc::zero = true,
+    )]
+    pub journal_entry: Account<'info, JournalEntryState>,
+    #[account(mut)]
+    pub owner: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+```
+
+In the above code, you should notice that it is very similar to creating a
+journal entry but there are a couple key differences. Since
+`update_journal_entry` is editing an already existing PDA, we do not need to
+`init`. However, the data being passed to the instruction could be different,
+we'll need to use the `realloc` constraint to reallocate the space for the
+account.
+
+- `realloc` - sets the new space required.
+- `realloc::payer` - defines the account that will either pay or be refunded
+  based on the newly required lamports
+- `realloc::zero` - defines that the account may be updated multiple times when
+  set to `true`
+
+The `seeds` and `bump` contraints are needed to be able to find the specific PDA
+we want to update.
+
```

Comment:
I think it would be worthwhile briefly describing the `mut` constraints in here too


```suggestion

The `mut` constraints allows us to mutate/change the data within the account.
Because how the Solana blockchain handles reading from accounts and writing to
accounts differently, we must explicitly define which accounts will be mutable
so the Solana runtime can correctly process them.
```